federationsender/query/api.go

```diff
@@ -0,0 +1,85 @@
+package query
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/matrix-org/dendrite/common"
+	"github.com/matrix-org/dendrite/federationsender/api"
+	"github.com/matrix-org/dendrite/federationsender/types"
+	rsAPI "github.com/matrix-org/dendrite/roomserver/api"
+	"github.com/matrix-org/util"
+)
+
+// FederationSenderQueryDatabase has the APIs needed to implement the query API.
+type FederationSenderQueryDatabase interface {
```

Comment:
We already have a storage interface, can we just use that? The only reason why we might want this would be for testing internal APIs in isolation without a database more easily (fewer things to implement) but we don't have any tests like that and I'd rather we tested the actual SQL as well.

Reply:
Yeah, don't see why not. This interface was there before. I'll remove it.